moabb/evaluations/evaluations.py

```diff
@@ -143,9 +147,71 @@ def _evaluate(self, dataset, pipelines):
                 for name, clf in run_pipes.items():
                     t_start = time()
                     cv = StratifiedKFold(5, shuffle=True, random_state=self.random_state)
-
+                    scorer = get_scorer(self.paradigm.scoring)
                     le = LabelEncoder()
                     y_cv = le.fit_transform(y[ix])
+                    X_ = X[ix]
+                    y_ = y[ix] if self.mne_labels else y_cv
+
+                    # ===============================================================================================
+                    # Implement Grid Search
+                    # ===============================================================================================
+                    # Load result if the folder exist
+                    name_grid = os.path.join(
+                        str(self.hdf5_path),
+                        "GridSearch_WithinSession",
+                        dataset.code,
+                        "subject" + str(subject),
+                        str(session),
+                        str(name),
+                    )
+                    if param_grid is not None and not os.path.isdir(name_grid):
+                        if name in param_grid:
+                            for k in np.arange(len(clf.steps)):
+                                for p in np.arange(len(list(param_grid[name].keys()))):
+                                    if (
+                                        clf.steps[k][0]
+                                        == (list(param_grid[name].keys())[p]).split(
+                                            sep="__"
+                                        )[0]
+                                    ):
+                                        search = GridSearchCV(
+                                            clf,
+                                            param_grid[name],
+                                            refit=True,
+                                            cv=cv,
+                                            n_jobs=-1,
+                                            scoring=self.paradigm.scoring,
+                                            return_train_score=True,
+                                        )
```

Comment:
It seems that you are conducting a grid search for all parameters separately. If you have: 
`param_grid["CSP+SVM"] = {"CSP__nfilters": [2, 6], "SVC__kernel": ["linear", "poly", "rbf"]}` 
You will make `GridSearchCV` for each steps (line 170), so you will run the GridSearch 2 times. I think you could get rid of the lines 170-177, running a grid search for each pipeline named in your param_grid.